### PR TITLE
Update and clean up SDK archive tools

### PR DESCRIPTION
**Pull request by parlough**
- Update dependencies
- Switch back to official platform detect now that it has a null safe release
- Switch to recommended lints
- Format and order Dart files
- Switch from pedantic's `unawaited` to an ignore comment
- Minor doc updates

### CODE REVIEW

File: src/tools/sdk/archive/pubspec.yaml [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 dependencies:
   dart_sdk_archive:
     path: ../dart_sdk_archive
+
 dev_dependencies:
   build_runner: ^2.0.0
-  pedantic: ^1.11.0
   build_web_compilers: ^3.0.0
+  lints: ^1.0.0
+  pedantic: ^1.11.0
```
Comment (kwalrath) [inline src/tools/sdk/archive/pubspec.yaml:14]: Do we really need pedantic?

File: src/tools/sdk/dart_sdk_archive/pubspec.yaml [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
 
 dev_dependencies:
-  build_runner: ^1.12.0
-  build_web_compilers: ^2.16.0
+  build_runner: ^2.0.0
+  build_web_compilers: ^3.0.0
   dart_style: ^2.0.0
   grinder: ^0.9.0
+  lints: ^1.0.0
   pedantic: ^1.11.0
```
Comment (kwalrath) [inline src/tools/sdk/dart_sdk_archive/pubspec.yaml:20]: Delete pedantic?

**Review by kwalrath — APPROVED** [review]
I see a couple of references to pedantic remain... Do we really need them? Otherwise, LGTM!

**Comment by kwalrath** [comment]
None of the code changes looks controversial, but I still hesitate to merge without an engineer taking a look.

**Comment by parlough** [comment]
@kwalrath We were using pedantic to mark unawaited futures, but since pedantic is deprecated and we can do that with comments instead, I've removed it in the latest commit.

File: src/tools/sdk/archive/_README.md [hunk 3]
```diff
@@ -6,4 +6,4 @@ If you modify that Dart code for this page, don't forget to regenerate
 the JavaScript:
 
     cd src/tools/sdk/archive
-    pub run build_runner build --release --output out
+    dart run build_runner build --release --output out
```
Comment (domesticmouse) [inline src/tools/sdk/archive/_README.md:9]: This works now? Or should it be `dart pub run build_runner build --release --output out` ?

File: src/tools/sdk/dart_sdk_archive/lib/src/run_app.dart [hunk 4]
```diff
@@ -32,7 +31,10 @@ Future<void> runApp() async {
     querySelector('#dev-os') as SelectElement,
   );
 
-  unawaited(stableSelector.init());
-  unawaited(betaSelector.init());
-  unawaited(devSelector.init());
+  // ignore: unawaited_futures
```
Comment (domesticmouse) [inline src/tools/sdk/dart_sdk_archive/lib/src/run_app.dart:34]: I have a vague recollection that `unawaited` was on it's way into the core libraries? @johnpryan probably knows this better than I do.

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
Just a couple of open questions, but not meant to be blocking

File: src/tools/sdk/archive/_README.md (hunk 3 above)
Comment (mit-mit) [reply src/tools/sdk/archive/_README.md:9]: Yes, this works

File: src/tools/sdk/dart_sdk_archive/lib/src/run_app.dart (hunk 4 above)
Comment (parlough) [reply src/tools/sdk/dart_sdk_archive/lib/src/run_app.dart:34]: Yep, but it won't be added until at least the next stable release, so don't want to rely on it yet. Here's a link to the new `unawaited` docs: https://api.dart.dev/dev/dart-async/unawaited.html

**Comment by parlough** [comment]
Rebased this on the latest arm64 -> ARM64 changes and waiting for tests to pass. Thanks for the reviews.

File: src/tools/sdk/archive/_README.md (hunk 3 above)
Comment (parlough) [reply src/tools/sdk/archive/_README.md]: This is gone over in more detail in the (newer) `dart run` page: https://dart.dev/tools/dart-run#in-a-depended-on-package

**Pull request merged**